packages/suite/src/middlewares/wallet/storageMiddleware.ts

```diff
@@ -0,0 +1,84 @@
+import { MiddlewareAPI } from 'redux';
+// import * as TRANSACTION from '@wallet-actions/constants/transactionConstants';
+import * as WALLET_SETTINGS from '@wallet-actions/constants/settingsConstants';
+// import * as transactionActions from '@wallet-actions/transactionActions';
+import * as storageActions from '@suite-actions/storageActions';
+import * as accountUtils from '@wallet-utils/accountUtils';
+import { SUITE } from '@suite-actions/constants';
+import { AppState, Action as SuiteAction, Dispatch } from '@suite-types';
+import { WalletAction } from '@wallet-types';
+import { ACCOUNT, DISCOVERY, TRANSACTION } from '@wallet-actions/constants';
+// import { ACCOUNT } from '@suite/actions/wallet/constants';
+// import { AccountInfo } from 'trezor-connect';
+
+const storageMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) => async (
+    action: SuiteAction | WalletAction,
+): Promise<SuiteAction | WalletAction> => {
+    // @ts-ignore
+    const prevState = api.getState();
+    // pass action
+    next(action);
+
+    switch (action.type) {
+        case SUITE.REMEMBER_DEVICE:
+            api.dispatch(storageActions.rememberDevice(action.payload));
+            break;
+
+        case ACCOUNT.CREATE:
+        case ACCOUNT.UPDATE: {
+            // TODO: explore better way to update accounts only
+            const device = accountUtils.getAccountDevice(api.getState().devices, action.payload);
+            if (device && device.features && device.remember) {
+                api.dispatch(storageActions.rememberDevice(device));
+            }
+            break;
+        }
+
+        case TRANSACTION.ADD:
+        case TRANSACTION.FETCH_SUCCESS: {
+            // TODO: explore better way to update accounts only
+            const device = accountUtils.getAccountDevice(api.getState().devices, action.account);
+            if (device && device.features && device.remember) {
+                api.dispatch(storageActions.rememberDevice(device));
+            }
+            break;
+        }
+
+        case TRANSACTION.REMOVE:
+            storageActions.removeAccountTransactions(action.account);
+            break;
+
+        case DISCOVERY.UPDATE:
+        case DISCOVERY.INTERRUPT:
+        case DISCOVERY.STOP:
+        case DISCOVERY.COMPLETE: {
+            // TODO: explore better way to update discovery only
+            const device = api.getState().devices.find(d => d.state === action.payload.deviceState);
```

Comment:
you can use suite.selectedDevice or instead, discovery is designed to run only on currently selected device

Reply:
wait :) i don't get why you are you saving device instead of discovery?

Reply:
`rememberDevice` should be triggered from `SUITE.REMEMBER_DEVICE`, right?

Reply:
+ additionaly device should be updated after DEVICE.UPDATE as well, for example changing label/FW/pin/passphrase settings